src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -91,8 +97,13 @@ private static boolean isValidPath(Path filePath) {
      *
      * @throws StorageOperationException if there were errors converting and/or storing data to file.
      */
-    public void save(AddressBook addressBook) throws StorageOperationException {
+    public void save(AddressBook addressBook) throws StorageOperationException, StorageFileNotFoundException {
 
+    	// check first whether the file exits or not
+    	if (!path.toFile().exists()) {
+    		throw new StorageFileNotFoundException("Addressbook storage file missing.");
```

Comment:
Correct check and throw.

Suggestion: The message should be defined as a constant for ease of modification in future, refer to how other exception error messages were coded.